README.md

```diff
@@ -95,6 +95,10 @@ Build Options
 | WITH_NEW_STRATEGIES      | --without-new-strategies | Use new strategies                                                                           | ON                               |
 | WITH_NATIVE_INSTRUCTIONS |                          | Instruct the compiler to use the full instruction set on this host (gcc/clang -march=native) | OFF                              |
 |                          | --force-sse2             | Assume SSE2 instructions are always available                                                | DISABLED (x86), ENABLED (x86_64) |
+| WITH_AVX2                |                          | Build with AVX2 slide hash                                                                   | ON                               |
+| WITH_SSE2                |                          | Build with SSE2 slide hash                                                                   | ON                               |
+| WITH_SSE4                |                          | Build with SSE4 CRC32-C                                                                      | ON                               |
+| WITH_PCLMULQDQ           |                          | Build with PCLMULQDQ CRC                                                                     | ON                               |
```

Comment:
I am thinking these options should not really be tied up to slide-hash or crc specifically, just let them be generic options to disable the instruction sets completely along with any and all features that depend on them. I do see that WITH_ACLE already refers to crc specifically, but I'd rather we avoid that.
Otherwise, we should name the options for example WITH_AVX2_SLIDE_HASH etc, and I don't think we need to have that level of granularity in the cmake files.

We could have more "hidden" checks for defines like DISABLE_AVX2_SLIDE_HASH to specifically disable one feature for debugging/testing (by short-circuiting the functable checks for example), but that is too verbose for CMake IMO.

Reply:
PCLMULQDQ and ACLE are both instruction sets that have limited use and such are exceptions, other are generic enough.

Reply:
@Dead2, are you saying I should just change the readme?

Reply:
@nmoinvaz Readme and the cmake descriptions, yes.